### PR TITLE
no-directive rule

### PR DESCRIPTION
**Pull request by Gillespie59**
Solve #271

### CODE REVIEW

**Comment by Gillespie59** [comment]
Any remarks on this new rule ? :D

File: rules/prefer-component.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                }
+            });
+        },
+        AssignmentExpression: function(node) {
+            // Only check for literal member property assignments.
+            if (node.left.type !== 'MemberExpression') {
+                return;
+            }
+            // Only check setting properties named 'link'.
+            if (node.left.property.name !== 'link') {
```
Comment (remcohaszing) [inline rules/prefer-component.js]: Shouldn't `compile`, `multiElement`, `priority`, `templateNamespace`,  `terminal` and `restrict` allowing `A` or `C` be allowed as well? (See https://docs.angularjs.org/api/ng/service/$compile#comprehensive-directive-api and https://docs.angularjs.org/guide/component#comparison-between-directive-definition-and-component-definition)

File: rules/prefer-component.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    return {
+        'angular:directive': function(callExpressionNode, fnNode) {
+            if (!fnNode || !fnNode.body) {
+                return;
+            }
+            fnNode.body.body.forEach(function(statement) {
+                if (statement.type === 'ReturnStatement' && !potentialReplaceNodes[statement.argument.name || '']) {
+                    context.report(statement, 'Directive should be implemented with the component method.');
```
Comment (remcohaszing) [inline rules/prefer-component.js]: Small performance boost: Use `fnNode.body.body.some` and `return true` here.

**Comment by Gillespie59** [comment]
Thanks for your feedback. I will update them asap

**Comment by Gillespie59** [comment]
PR merged

**Pull request closed without merging**